packages/domain/src/use-cases/publications/CreateComment.ts

```diff
@@ -22,27 +22,21 @@ export type CreateCommentRequest = {
    * The metadata URI.
    */
   metadata: URI;
-} & (
-  | {
-      /**
-       * Host on Momoka
-       */
-      momoka: true;
-    }
-  | {
-      /**
-       * Host on on-chain
-       */
-      momoka: false;
-      /**
-       * The Open Actions associated with the publication.
-       */
-      actions?: NonEmptyArray<OpenActionConfig>;
-      /**
-       * The post reference policy.
-       */
-      reference: ReferencePolicyConfig;
-    }
-);
+  /**
+   * The Open Actions associated with the publication.
+   */
+  actions?: OpenActionConfig[];
+  /**
+   * The post reference policy.
+   */
+  reference?: ReferencePolicyConfig;
+};
 
-export class CreateComment extends MomokaOption<CreateCommentRequest> {}
+export class CreateComment extends MomokaCapable<CreateCommentRequest> {
+  override async execute(request: CreateCommentRequest): Promise<void> {
+    if (['actions', 'reference'].some((key) => key in request)) {
```

Comment:
so that's how we know if post should go onchain or to momoka?

Reply:
Yeah, i double checked this with Josh who said we should abstract this away from end users and use momoka whenever it makes sense.